docs/apps/index.md

```diff
@@ -0,0 +1,96 @@
+# App object for ADK
+
+<div class="language-support-tag">
+    <span class="lst-supported">Supported in ADK</span><span class="lst-python">Python v1.14.0</span>
+</div>
+
+The `App` class is a top-level container for an entire Agent Development Kit
+(ADK) agent workflow. It is designed to manage the lifecycle, configuration, and
+state for a collection of agents. The App class separates the concerns of an
+agent workflow's overall operational infrastructure from individual agents'
+task-oriented reasoning. 
+
+Defining an App object in your ADK workflow is optional and changes how you
+organize your agent code and run your agents. From a practical perspective, you
+use the App class to configure the following features for your agent workflow:
+
+*   [**Context caching**](/adk-docs/context/caching/)
+*   [**Context compression**](/adk-docs/context/compaction/)
+*   [**Agent resume**](/adk-docs/runtime/resume/)
+*   [**Plugins**](/adk-docs/plugins/)
+
+This guide explains how to use the App class for configuring and managing your
+ADK agent workflows.
+
+## Use the App class
+
+The App class is used as the primary container of your agent workflow and
+contains the root agent of the project.
+
+### Define your Agent
+
+First, define an agent using a class derived from BaseAgent. The
+following code sample defines a simple greeter agent:
+
+```python
+from google.adk.agents import Agent
+
+class GreeterAgent(Agent):
+    name = "Greeter Agent"
+    description = "An agent that provides a friendly greeting."
+
+    def __call__(self, unused_input: str) -> str:
+        return "Hello, world!"
+
+```
+
+### Define your App
+
+Create a class that inherits from `App`. Configure this object with the
+`root_agent` parameter and optional features, as shown in the following
+sample code:
+
+```python
+from google.adk.apps import App
+
+class MyGreeterApp(App):
+    name = "Simple Greeter App"
+    root_agent = GreeterAgent()
+    # Optionally include App-level features:
+    # plugins, context_cache_config, resumability_config
+
+my_app = MyGreeterApp()
```

Comment:
it should be exported as app = ... (convention) so it works with adk-web and a few other things. (like the convention of root_agent)

Reply:
got it. updated and noted "app" convention